src/bin/main.rs

```diff
@@ -136,12 +172,51 @@ fn check(args: &ArgMatches) -> HowserResult<Option<ValidationProblem>> {
     }
 }
 
+fn process_pharmacy_file(args: &ArgMatches) -> HowserResult<Vec<ValidationProblem>> {
+    let mut report: Vec<ValidationProblem> = Vec::new();
+
+    if let Some(filename) = args.value_of("pharmacy") {
+        if let Ok(pharmacy_contents) = get_file_contents(filename) {
+            if let Some(specs) = pharmacy_contents.parse::<Value>()?["Specs"].as_table() {
+                for rx_file in specs.keys() {
+                    if let Some(doc_file) = specs.get(rx_file).map_or(None, |value| value.as_str())
```

Comment:
Rather than getting the iterating through the keys and then later getting the values, it seems like it would be less redundant/error-prone to instead iterate through the entries (key-value pairs).

```rust
for (key, value) in specs.iter() {
    if let Some(doc_file) = value.as_str() {
        ...
    }
}
```

Separately, it seems like there's no remediation or identification of a problem if the `value` is not a string.